utilities/BuildUtilities.groovy

```diff
@@ -370,3 +371,98 @@ def createDatasets(String[] datasets, String options) {
 	}
 }
 
+/*
+ * getFileProperty - verifies if a properties file for the buildFile exists to overwrite properties 
+ */
+def getFileProperty(String prop, String buildFile){
+	// validate if build framework allows overwrites
+	if (props.allowFileLevelOverwrites && props.allowFileLevelOverwrites.toBoolean()){
+		if(!fileLevelPropertiesCache.containsKey(buildFile)){
+			loadFileLevelOverwrites(buildFile)
+		}
+
+		fileLevelOverwrites = fileLevelPropertiesCache.get(buildFile)
+		if (fileLevelOverwrites != null) {
+			fileLevelProperty = fileLevelOverwrites.getProperty(prop)
+			if (fileLevelProperty!=null){
+				if (props.verbose) println("** File level overwrite found for $buildFile - $prop: $fileLevelProperty")
+				if (props.verbose)
+					return fileLevelProperty
+			}
+		}
+	}
+
+	//return DBB File Property or Global property
+	return props.getFileProperty(prop, buildFile)
+}
+
+/*
+ * getFileProperty - verifies if a properties file for the buildFile exists to overwrite properties 
+ */
+def getFileProperty(String prop, String buildFile){
+	// validate if build framework allows overwrites
+	if (props.allowFileLevelOverwrites && props.allowFileLevelOverwrites.toBoolean()){
+		if(!fileLevelPropertiesCache.containsKey(buildFile)){
+			loadFileLevelOverwrites(buildFile)
+		}
+
+		fileLevelOverwrites = fileLevelPropertiesCache.get(buildFile)
+		if (fileLevelOverwrites != null) {
+			fileLevelProperty = fileLevelOverwrites.getProperty(prop)
+			if (fileLevelProperty!=null){
+				if (props.verbose) println("** File level overwrite found for $buildFile - $prop: $fileLevelProperty")
+				// check if overwrite is allowed
+				if (props.whitelistFileLevelOverwrites.split(",").contains(prop))
+					return fileLevelProperty
+				else
+				if (props.verbose) println("*! Overwrite for $prop not allowed. Not on whitelist. See build property whitelistFileLevelOverwrites.")
+			}
+		}
+	}
+
+	//return DBB File Property or Global property
+	return props.getFileProperty(prop, buildFile)
+}
+
+/*
+ * groovy metaprogramming to return properties to language scripts
+ *  - verifies if a propertiesfile for the buildFile exists and overwrites settings
+ */
+static def propertyMissing(String name) {
+	return props.getProperty(name)
+}
+
```

Comment:
You introduce a behavior here that can complexity the debugging. This simple test case raise no error for balibalo.

```
@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
import com.ibm.dbb.build.*
import groovy.transform.*

// define script properties
@Field static BuildProperties props = BuildProperties.getInstance()

balibalo

/*
 * groovy metaprogramming to return properties to language scripts
 *  - verifies if a propertiesfile for the buildFile exists and overwrites settings
 */
static def propertyMissing(String name) {
        return props.getProperty(name)
}

```
```
>$DBB_HOME/bin/groovyz test.groovy
** Build finished
```

You can add an assert if `props.getProperty(name)`is null.

Reply:
Hi @jbyibm, you are right.

Already today, there are several fields which are used and are tested against NULL.

Additionally, `out` seems to be an object being returned when creating a new instance of an object ... A bit odd ... 
 

```
static def propertyMissing(String name) {
	def optionalParms = ["out", "errPrefix", "debug", "runzTests", "team", "logEncoding", "RESLIB", "userBuild"]
	if (!optionalParms.contains(name)) assert props."$name" : "*! Missing property / field '$name'"
	return props.getProperty(name)
}
```